### PR TITLE
Commit linting

### PR DESCRIPTION
**Pull request by coopbri**
##### Task link: https://trello.com/c/O7zkMZIX/41-commit-linting-hook

- Added commit linting via [commitlint](https://commitlint.js.org/#/) with Husky-enabled hook (`commit-msg`)
- Updated lifecycle hook scripts to work for Yarn Berry (based on [Husky Yarn docs](https://typicode.github.io/husky/getting-started.html#yarn-2))
- Explicitly set introduction MDX as first in story ordering (see [this comment](https://github.com/animareflection/ui/pull/56/files#r1304610536))

## Test Steps

- Locally, test `commitlint` and make sure it leverages `commitlint.config.ts`
  - This can be tested by piping input to the `commitlint` CLI. For example, `echo 'test: example' | yarn commitlint`
- Verify the Husky `commit-msg` hook automatically executes `commitlint`
  - This can be tested by adding a random file (don't commit it, just `git add`), then testing various git messages with `git commit`
  - CLI & config unit tests will be added later once this config is extracted to a shared repo (checklist item added to https://trello.com/c/O7zkMZIX/41-commit-linting-hook scope)
- Verify the `commitlint.config.ts` config looks good, particularly the commit types
- Verify introduction story is loaded first on page load

### CODE REVIEW

File: commitlint.config.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { RuleConfigSeverity } from "@commitlint/types";
+
+import type { UserConfig as CommitlintUserConfig } from "@commitlint/types";
+
+/**
+ * `commitlint` configuration.
+ * @see https://commitlint.js.org/#/reference-configuration
+ */
+const commitlintConfig: CommitlintUserConfig = {
```
Comment (coopbri) [inline commitlint.config.ts:9]: Unit tests will be added to test the CLI against this config (added to https://trello.com/c/O7zkMZIX/41-commit-linting-hook scope)

File: .storybook/main.ts [hunk 2]
```diff
@@ -14,7 +14,11 @@ const storybookConfig: StorybookConfig = {
     name: "@storybook/nextjs",
     options: {},
   },
-  stories: ["../src/**/*.stories.@(ts|tsx|mdx)"],
+  stories: [
+    // NB: intentional pattern duplication (with capture group below) to ensure introduction page is loaded first
+    "../src/stories/introduction.stories.mdx",
```
Comment (coopbri) [inline .storybook/main.ts:19]: Without this, the first CSF story (Accordion at time of writing) is loaded on initial page load

File: package.json [hunk 3]
```diff
@@ -52,7 +52,9 @@
     "storybook:build": "panda codegen && storybook build",
     "changeset": "changeset",
     "release": "yarn build && changeset publish",
-    "prepare": "husky install"
+    "postinstall": "husky install",
+    "prepack": "pinst --disable",
+    "postpack": "pinst --enable"
```
Comment (hobbescodes) [inline package.json:57]: Nice! Appreciate you updating this! Verified that `postinstall` script did run appropriately after installing deps locally.

File: commitlint.config.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    // specify valid types
+    "type-enum": [
+      RuleConfigSeverity.Error,
+      "always",
+      [
+        "build", // build system, dependencies, tooling
+        "chore", // maintenance
+        "ci", // continuous integration
+        "docs", // update documentation
+        "feat", // add a feature (option 1 of 2)
```
Comment (hobbescodes) [inline commitlint.config.ts:24]: Also appreciate you adding `feat` even though you really didn't need to lol.

Comment (coopbri) [reply commitlint.config.ts:24]: Np! Closer in alignment to the [Conventional Commits spec](https://www.conventionalcommits.org/en/v1.0.0/) anyways!

**Review by hobbescodes — APPROVED** [review]
Everything lgtm! Verified all test steps! Introduction does get loaded first on initial page load for storybook now, `commitlint` config looks great and linting worked great! Great job!

**Comment by Twonarly1** [comment]
Storybook Introduction is not loading first. Still jumps to Accordion component.

**Comment by Twonarly1** [comment]
> Everything lgtm! Verified all test steps! Introduction does get loaded first on initial page load for storybook now, `commitlint` config looks great and linting worked great! Great job!

Storybook still loading Accordion for me.

**Comment by hobbescodes** [comment]
> Storybook Introduction is not loading first. Still jumps to Accordion component.

Hmmm... worked for me. You want to try and remove `node_modules` + and reinstall deps (or even just remove the `.cache` directory inside `node_modules`). Then make sure that you are initially loading `localhost:6006` with no additional path appended.

**Comment by Twonarly1** [comment]
Verified storybook loads introduction on page load. Needed to pull changes again.

**Review by Twonarly1 — APPROVED** [review]
Looks good. Husky commit-msg fails on wrong entry. Storybook introduction loads initially.

**Pull request merged**